src/main/java/net/minecraftforge/event/ForgeEventFactory.java

```diff
@@ -635,5 +635,13 @@ public static boolean canCreateFluidSource(World world, BlockPos pos, IBlockStat
         Result result = evt.getResult();
         return result == Result.DEFAULT ? def : result == Result.ALLOW;
     }
+    
+    public static boolean onPistonMove(World world, BlockPos pos, EnumFacing facing, boolean extending)
+    {
+        if (extending)
+            return MinecraftForge.EVENT_BUS.post(new net.minecraftforge.event.world.PistonEvent.PistonExtendEvent(world, pos, facing));
+        else
+            return MinecraftForge.EVENT_BUS.post(new net.minecraftforge.event.world.PistonEvent.PistonRetractEvent(world, pos, facing));
```

Comment:
Please import `PistonEvent`, this is not a patch so you can import anything you need to.